java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -3024,12 +3035,16 @@ public long getNumRows() {
 
   @Override
   public int getNumChildren() {
-    if (getDataType() != DType.LIST /* or structs */) {
+    if (getDataType() != DType.LIST && getDataType() != DType.STRUCT) {
       return 0;
     }
     return offHeap.getNumChildren(getNativeView());
   }
 
+  private boolean isNestedType() {
+    return type == DType.LIST || type == DType.STRUCT;
```

Comment:
I would prefer to have the DType indicate if it is a nested type or not, so it is more maintainable.

Reply:
Sure, I like that, let me fix that up

Reply:
Done.